pallets/foreign-investments/src/lib.rs

```diff
@@ -72,9 +68,8 @@ pub type ForeignId<T> = (
 /// Swap alias
 pub type SwapOf<T> = Swap<<T as Config>::SwapBalance, <T as Config>::CurrencyId>;
 
-/// Swap state alias
-pub type SwapStateOf<T> =
-	SwapState<<T as Config>::SwapBalance, <T as Config>::SwapBalance, <T as Config>::CurrencyId>;
+/// Identification of a swap from foreing-investment perspective
+pub type SwapId<T> = (<T as Config>::InvestmentId, Action);
```

Comment:
Note that the meaning of `SwapId` has changed regarding the previous implementation (see the description of the PR)

Reply:
**MAYBE CR**

Aren't we missing a migration for the existing storage to the one used in this case here?